api/proto/config/appconfig.proto

```diff
@@ -26,6 +26,47 @@ enum MetaDataType {
   MetaDataDriveMultipart = 3; // Process multipart MIME for application
 }
 
+// The information about a snapshot
+message AvailableSnapshot {
+  string id = 1;
+}
+
+// The snapshot information for an Application Instance
+message SnapshotConfig {
+  // activeSnapshot is the id of the snapshot which is currently used.
+  // It may be empty if no snapshot is used. Worth to note that the field is
+  // not used alone to trigger a rollback. It just indicates that the device
+  // is currently using a snapshot with the given id. To trigger a rollback, in
+  // addition to it, the rollbackCounter should be increased. It is done to
+  // avoid unnecessary rollbacks when the same activeSnapshot is sent several
+  // times (we cannot relay on EVE to perform rollbacks after the first time
+  // it receives the corresponding message, so it's a regular case to send the
```

Comment:
What you're doing with the counter is correct, but the description in terms of messages is confusing.
The API contains a declarative statement for the intended config which is replicated from the controller to the device; some times the device can be behind because it hasn't yet managed to fetch the latest declarative statement.
This implies that for operations (like a device reboot or app instance restart) we use a counter, and an increase in the declarative counter value has the effect that EVE will perform some operation.
I think it would be good to use that way to describe it (but don't need all of the above verbiage); you can look at how the reboot and restart are described with their counter in the *.proto file.

Reply:
Ok, I'll try to reword it.

Reply:
Done